data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/StandardExtensionProvider.java

```diff
@@ -0,0 +1,33 @@
+package org.opensearch.dataprepper.plugins.sink.s3;
+
+import org.opensearch.dataprepper.model.codec.OutputCodec;
+import org.opensearch.dataprepper.plugins.sink.s3.compression.CompressionOption;
+
+class StandardExtensionProvider implements ExtensionProvider {
+    private final String extension;
+
+    static ExtensionProvider create(OutputCodec outputCodec, CompressionOption compressionOption) {
+
+        String codecExtension = outputCodec.getExtension();
+
+        if(outputCodec.isCompressionInternal()) {
+            return new StandardExtensionProvider(codecExtension);
+        }
+
+        String extension = compressionOption.getExtension()
```

Comment:
Will the configured extension be ignored if there is a internal compression?

Reply:
Yes, it will be ignored.

With internal compression the file itself is not compressed. Different parts are independently compressed. A user should not try to run it through GZip/Snappy decompression. 

With Parquet specifically, the compression is stored as metadata inside the file. And compressed files retain the `.parquet` extension.

Reply:
This test verifies that exact behavior: `getExtension_returns_extension_of_codec_when_compression_internal`.